### DISCUSSION TITLE
Cannot find dependencies of library built with parcel

### DISCUSSION THREAD

**Discussion by tillsanders** [parcel-bundler/parcel#4742] [General] [answered]
I author a library that runs in the browser. It is, however, part of a bigger application. Therefore I build it with `--target=node`, so the dependencies are not included in the build. I then import this library in my application. This application is bundled with parcel and the default target. For some reason, parcel is unable to locate any of the dependencies of my library, even if I try to install them in the scope of my application additionally. What's also confusing, is that the error message complaining about this doesn't appear during compile time in the console, but only during runtime in the browser's console. I suspect that I'm misunderstanding some concept here... any ideas?

I'm using the latest parcel v1.

**Comment by mischnic** [comment, marked as answer]
The problem is that the output of Parcel 1 always looked like this:
```js
(function (modulesMap, entry) {
  // internal runtime
})(
  {
    "index.js": function (require, module, exports) {
      // contents of `src/index.js`
      var Foo = require("./thing.js");
      var obj = new Foo();
      obj.run();
    },
    "thing.js": function (require, module, exports) {
      // contents of `src/thing.js`
      var lodash = require("lodash");
      module.exports = class Foo {
        run() {
          console.log(lodash.add(1, 2));
        }
      };
    },
  },
  "index.js"
);
```

But if you try to run Parcel over this again, it cannot infer that `require("lodash")` should include the lodash package because `require` appears to be a local variable.

Bundles generated by Parcel 2  don't have this problem (because the individual file contents aren't grouped into functions but concatenated together via scope-hoisting).

**Reply by tillsanders** [reply]
Ahh, I see! And is there a way to use Parcel 1 in my setup?

**Reply by mischnic** [reply]
Building the library with `--experimental-scope-hoisting` should solve this problem.
Unless you run into bugs due to its experimental nature...
